activerecord/lib/active_record/store.rb

```diff
@@ -86,29 +91,32 @@ class << self
     module ClassMethods
       def store(store_attribute, options = {})
         serialize store_attribute, IndifferentCoder.new(store_attribute, options[:coder])
-        store_accessor(store_attribute, options[:accessors], prefix: options[:prefix]) if options.has_key? :accessors
+        store_accessor(store_attribute, options.delete(:accessors), options) if options.has_key? :accessors
       end
 
-      def store_accessor(store_attribute, *keys, prefix: nil)
-        keys = keys.flatten
-
-        accessor_prefix =
-          case prefix
-          when String, Symbol
-            "#{prefix}_"
-          when TrueClass
-            "#{store_attribute}_"
-          else
-            ""
-          end
+      def store_accessor(store_attribute, *args)
+        options = args.extract_options!
+        keys = args.flatten
 
         _store_accessors_module.module_eval do
           keys.each do |key|
-            define_method("#{accessor_prefix}#{key}=") do |value|
+            prefix = if options[:prefix] == true
```

Comment:
The prefix/suffix is just enough calculated once, there is no reason to do for each keys.